homeassistant/components/esphome/sensor.py

```diff
@@ -52,9 +58,22 @@ def _state(self) -> Optional[SensorState]:
     @property
     def icon(self) -> str:
         """Return the icon."""
-        if not self._static_info.icon or self._static_info.device_class:
+        if self._static_info.device_class:
             return None
-        return self._static_info.icon
+        icon_validator = vol.Or(cv.whitespace, cv.icon)
+        try:
+            icon_validated = icon_validator(self._static_info.icon)
+        except vol.Invalid as ex:
```

Comment:
Probably just let it blow up instead since this should be caught during development when upgrading the library.

Reply:
No problem, I removed the icon string validation.

Reply:
I meant that we should validate the icon but let it raise an error if the validation fails. Without validation, an incorrect icon is not sure to raise an error clearly in the backend.

Reply:
Thanks, now I see, I'll keep the validation and only remove the try/catch phrase.